### PR TITLE
Hints for user mistyped table name or database name.

### PR DESCRIPTION
**Pull request by fastio**
Changelog category (leave one):
- Improvement
 

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Hints for table name or database name. #35450 

> Information about CI checks: https://clickhouse.tech/docs/en/development/continuous-integration/

### CODE REVIEW

File: src/Interpreters/DatabaseCatalog.cpp [hunk 1]
```diff
@@ -1083,6 +1092,27 @@ DDLGuard::DDLGuard(Map & map_, std::shared_mutex & db_mutex_, std::unique_lock<s
...
+        DatabasePtr database = DatabaseCatalog::instance().tryGetDatabase(database_name);
+        if (!database)
+            continue;
+
+        for (auto table_it = database->getTablesIterator(getContext()); table_it->isValid(); table_it->next())
+        {
+            const auto & storage_id = table_it->table()->getStorageID();
+            result.emplace_back(storage_id.getFullTableName());
+        }
+    }
```
Comment (tavplubix) [inline src/Interpreters/DatabaseCatalog.cpp:1112]: I think it would be better to ad separate hints for database name and for table name

**Review by tavplubix — COMMENTED** [review]
Could you please add a test?

File: src/Interpreters/DatabaseCatalog.cpp (hunk 1 above)
Comment (fastio) [reply src/Interpreters/DatabaseCatalog.cpp]: That's good idea. I have fixed it.

**Comment by fastio** [comment]
>

**Comment by fastio** [comment]
> Could you please add a test?

Sure.

**Pull request closed without merging**